components/suggest/src/geoname.rs

```diff
@@ -207,8 +232,42 @@ struct DownloadedGeonamesAlternatesAttachment {
     /// code: "en", "de", "fr", etc. Can also be a geonames pseduo-language like
     /// "abbr" for abbreviations and "iata" for airport codes.
     language: String,
-    /// Tuples of geoname IDs and their localized names.
-    names_by_geoname_id: Vec<(GeonameId, Vec<String>)>,
+    /// Tuples of geoname IDs and their alternate names.
+    alternates_by_geoname_id: Vec<(GeonameId, Vec<DownloadedGeonamesAlternate<String>>)>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(untagged)]
+enum DownloadedGeonamesAlternate<S: Deref<Target = str>> {
```

Comment:
In general, `<S: AsRef<str>>` is more flexible than `<S: Deref<Target = str>>`. As a result, the former trait constraint would cover more types than the latter. This [comment](https://users.rust-lang.org/t/understanding-the-perils-of-deref/47958/7) summarizes the differences and subtleties quite well.  
```suggestion
enum DownloadedGeonamesAlternate<S: AsRef<str>> {
```